Libraries/VirtualFunctions/VirtualFunctions.h

```diff
@@ -0,0 +1,312 @@
+  virtual_map[current_row - 2][current_column] != '3') {
+    virtual_map[current_row - 2][current_column] = 'R';
+  }
+  current_row -= 2;
+}
+
+// Moves the position of the robot on the virtual map to the right.
+void moveMapEast() {
+  virtual_map[current_row][current_column] = 'V';
+  if (virtual_map[current_row][current_column + 2] != '1'
+  && virtual_map[current_row][current_column + 2] != '2'
+  && virtual_map[current_row][current_column + 2] != '3') {
+    virtual_map[current_row][current_column + 2] = 'R';
+  }
+  current_column += 2;
+}
+
+// Moves down the position of the robot on the virtual map.
+void moveMapSouth() {
+  virtual_map[current_row][current_column] = 'V';
+  if (virtual_map[current_row + 2][current_column] != '1'
+  && virtual_map[current_row + 2][current_column] != '2'
+  && virtual_map[current_row + 2][current_column] != '3') {
+    virtual_map[current_row + 2][current_column] = 'R';
+  }
+  current_row += 2;
+}
+
+// Moves the position of the robot on the virtual map to the left.
+void moveMapWest() {
+  virtual_map[current_row][current_column] = 'V';
+  if (virtual_map[current_row][current_column - 2] != '1' && virtual_map[current_row][current_column - 2] != '2' && virtual_map[current_row][current_column - 2] != '3') {
+    virtual_map[current_row][current_column - 2] = 'R';
+  }
+  current_column -= 2;
+}
+
+// Function that prints the robot's intern map.
+void printMap(Map tiles_map, const uint8_t iFloor) {
+  char char_map[tiles_map.numberOfRows()*2 + 1][tiles_map.numberOfColumns()*2 + 1];
+  int row_char, column_char;
+
+  for(uint8_t i = 0; i < tiles_map.numberOfRows()*2 + 1; ++i) {
+    for(uint8_t j = 0; j < tiles_map.numberOfColumns()*2 + 1; ++j) {
+      char_map[i][j] = '.';
+    }
+  }
+  
+  row_char = 1;
+
+  for (int row_tile = 0 ; row_tile < tiles_map.numberOfRows() ; ++row_tile) {
+    column_char = 1;
+    for (int column_tile = 0 ; column_tile < tiles_map.numberOfColumns() ; ++column_tile) {
+      if(tiles_map.getTile(row_tile, column_tile).getStart()) {
+        char_map[row_char][column_char] = 'S';
+      }
+      else if(tiles_map.getTile(row_tile, column_tile).isVisited()) {
+        char_map[row_char][column_char] = 'V';
+      }
+      else if(tiles_map.getTile(row_tile, column_tile).isAccessible()) {
+        char_map[row_char][column_char] = ' ';
+      }
+      else {
+        char_map[row_char][column_char] = 'X';
+      }
+      if(tiles_map.getTile(row_tile, column_tile).getWeight() == 5) {
+        char_map[row_char][column_char] = '1';
+      }
+      else if(tiles_map.getTile(row_tile, column_tile).getWeight() == 9) {
+        char_map[row_char][column_char] = '2';
+      }
+      else if(tiles_map.getTile(row_tile, column_tile).getWeight() == 13) {
+        char_map[row_char][column_char] = '3';
+      }
+      else if(tiles_map.getTile(row_tile, column_tile).isBlack()) {
+        char_map[row_char][column_char] = 'B';
+      }
```

Comment:
I'd recommend to separate into individual functions returning a character each bunch of:
if {
} else if {
} else {
}

So it's cleaner  and at the end you will have a few functions instead of many if else in here.

Reply:
Also, aren't you overwriting the values of char_map? Don't you assign an X and then maybe a 1 or 2 or 3? Or is that the desired behaviour?

Reply:
Ok.